match2/commons/match_group_display_bracket.lua

```diff
@@ -1,738 +1,745 @@
-    local nilSpacer = {NIL_EXTRADATA, nil}
-
-    local newdepth = depth + 1 + self.skipround
-
-    local vals1, s1 =
-        unpack(
-        upper ~= nil and
-            upper:buildBracket(
-                firsthead,
-                true,
-                (lower ~= nil and 1 or 0) - 1,
-                false,
-                newdepth,
-                maxdepth,
-                singledepth,
-                headerchild
-            ) or
-            nilSpacer
-    )
-    local height1, singleDepth1, endSingledepth1, midCorrection1 = unpack(vals1)
-    local vals2, s2 =
-        unpack(
-        lower ~= nil and
-            lower:buildBracket(
-                firsthead,
-                false,
-                (upper ~= nil and -1 or 0) + 1,
-                false,
-                newdepth,
-                maxdepth,
-                singledepth,
-                issingle and headerchild
-            ) or
-            (upper ~= nil and nilSpacer or endSpacer)
-    )
-    local height2, singleDepth2, endSingledepth2, midCorrection2 = unpack(vals2)
-
-    local ownHeight =
-        self.height + self.padding + (headerchild and self.headerHeight2 or 0) +
-        math.max(singledepth, self.qualwin and self.quallose and 1 or 0) * self.stepHeight
-    local endSingleDepth = isend and singledepth or (upper == nil and endSingledepth2 or 0)
-    local uppermid = height1 / 2 - midCorrection1
-    local upperinset = height1 - uppermid
-    local lowermid = height2 / 2 + midCorrection2
-
-    -- adjust mid position for two child matches/one child match/no child matches
-    local midCorrection = 0
-    if (isfull) then
-        midCorrection = (uppermid - lowermid) / 2
-    elseif (issingle) then
-        midCorrection = -self.stepHeight
-    elseif (isend) then
-        midCorrection =
-            (headerchild and self.headerHeight2 / 2 or 0) +
-            (singledepth - (self.qualwin and self.quallose and 1 or 0)) * self.stepHeight / 2
+    if match.bracketData.qualWin then
+        headerNode:node(
+            Bracket.MatchHeader({
+                header = config.qualifiedHeader or '!q',
+                height = config.headerHeight,
+            })
+                :addClass('brkts-qualified-header')
+                :css('--qual-skip', match.bracketData.qualSkip)
+        )
     end
 
-    midCorrection = midCorrection + midCorrection1 + midCorrection2
-    maxHeight = math.max(ownHeight, height1 + height2)
-    lineheight = maxHeight / 2 + midCorrection - self.padding - upperinset
-    local marginTop = maxHeight / 2 
-        - self.height / 2 
-        - (headerchild and self.headerHeight + 8 or 0) 
-        + midCorrection
-
-    local op1, op2 = unpack(_createOpponentData(self, bracketreset))
-    local opponent1node, opponent1hash, tbd1 = unpack(op1)
-    local opponent2node, opponent2hash, tbd2 = unpack(op2)
-    
-    local hasDetails = WikiSpecific.matchHasDetails(self)
+    return headerNode
```

Comment:
Decent candidate for a function

Reply:
Since it's just passing props to `MatchHeader`, and adding a css class and style, I think it's too small to be a separate function. In terms of functionality it's too similar to `MatchHeader`. It would be used in only one place.